### PR TITLE
Buzzer implementation to drive the buzzer

### PR DESCRIPTION
**Pull request by Coscolin**
Add buzzer implementation with this fixed values:

- In platformio.ini:
```
BUZZER_PIN=13         ; ESP32 pin GPIO13 connected to piezo buzzer
BUZZER_HYSTERESIS=50  ; Hysteresis PPM to avoid BUZZER ON and OFF continuously if repeat is once

```
- In CO2_Gadget.ino:
```
bool activeBuzzer = true;
bool repeatBuzzer = true;
uint16_t toneBuzzerBeep = 1000;
uint16_t durationBuzzerBeep = 100;
uint16_t timeBetweenBuzzerBeep = 10;
uint64_t lastTimeBuzzerBeep = 0;  // Time of last Buzzer loop

```

When inMenu stop buzzer.

When CO2 values rise co2OrangeRange active buzzer twice (bep-bep).
When CO2 values rise co2RedRange active buzzer three times (bep-bep-bep).

TODO:

- [ ] Implement menu configuration
- [ ] Implement web configuration
- [ ] Save values to preferences

Review logical flow of buzzerLoop(). Now only check co2 values rise limit each timeBetweenBuzzerBeep.

### CODE REVIEW

**Comment by melkati** [comment]
Comes from #153 more info and discussion there

File: CO2_Gadget.ino [hunk 1]
```diff
@@ -61,6 +61,13 @@ uint16_t measurementInterval = 10;
 bool bleInitialized = false;
 int8_t selectedCO2Sensor = -1;
 bool outputsModeRelay = false;
+bool activeBuzzer = true;
```
Comment (melkati) [inline CO2_Gadget.ino:64]: Should be false by default

File: CO2_Gadget.ino [hunk 2]
```diff
@@ -240,13 +247,23 @@ uint16_t batteryFullyChargedMillivolts = 4200;  // Voltage of battery when it is
 #include "CO2_Gadget_TFT.h"
 #endif
 
+/*****************************************************************************************************/
+/*********                                                                                   *********/
+/*********                         INCLUDE MENU FUNCIONALITY                                 *********/
```
Comment (melkati) [inline CO2_Gadget.ino:252]: Is buzzer

File: CO2_Gadget.ino [hunk 3]
```diff
@@ -513,4 +530,7 @@ void loop() {
     buttonsLoop();
     menuLoop();
     BLELoop();
+    #if defined SUPPORT_BUZZER
```
Comment (melkati) [inline CO2_Gadget.ino:533]: Defines must go to CO2_Gadget_Buzzer.h, as others, to keep the main loop clean

**Review by melkati — CHANGES_REQUESTED** [review]

**Comment by melkati** [comment]
I miss the define in platformio.ini to use buzzer activation by HIGH or LOW

File: CO2_Gadget.ino (hunk 1 above)
Comment (Coscolin) [reply CO2_Gadget.ino:64]: As this implementation is done without menu configuration, if you set activeBuzzer = false you can't test it.

**Comment by melkati** [comment]
To make the buzzer code non-blocking and eliminate delays, you can utilize the `millis()` function for timing and avoid using `delay()`. Below is the modified code:

```cpp
uint64_t lastTimeBuzzerBeep = 0;  // Time of last Buzzer loop

bool belowOrangeRange = true;
bool belowRedRange = true;

void wakeUpDisplay() {
    if (actualDisplayBrightness == 0)  // Turn on the display only if it's OFF
    {
#if defined(SUPPORT_OLED) || defined(SUPPORT_TFT)
        setDisplayBrightness(DisplayBrightness);  // Turn on the display at DisplayBrightness brightness
#endif
        lastTimeButtonPressed = millis();
    }
    return;
}

void buzzerRedRange() {
    Serial.println("[BUZZ] Buzzer RED range");
    tone(BUZZER_PIN, toneBuzzerBeep + co2, durationBuzzerBeep);
}

void buzzerOrangeRange() {
    Serial.println("[BUZZ] Buzzer ORANGE range");
    tone(BUZZER_PIN, toneBuzzerBeep + co2, durationBuzzerBeep);
}

void buzzerLoop() {
    if (!activeBuzzer)
        return;

    if (inMenu) {  // Inside Menu stop BEEPING
        noTone(BUZZER_PIN);
        return;
    }

    if ((millis() - lastTimeBuzzerBeep >= timeBetweenBuzzerBeep * 1000) || (lastTimeBuzzerBeep == 0)) {
        lastTimeBuzzerBeep = millis();

        if (co2 > co2RedRange) {
            if (belowRedRange || repeatBuzzer) {
                wakeUpDisplay();
                buzz.once(0, buzzerRedRange);
                belowRedRange = false;
            }
            return;
        } else if (co2 < (co2RedRange - BUZZER_HYSTERESIS)) {
            belowRedRange = true;
        }

        if (co2 > co2OrangeRange) {
            if (belowOrangeRange || repeatBuzzer) {
                wakeUpDisplay();
                buzz.once(0, buzzerOrangeRange);
                belowOrangeRange = false;
            }
            return;
        } else if (co2 < (co2OrangeRange - BUZZER_HYSTERESIS)) {
            belowOrangeRange = true;
        }
    }
}
```

In this modified code:

- The `millis()` function is used to track the time for non-blocking timing.
- The `delay()` calls within the `buzzerRedRange()` and `buzzerOrangeRange()` functions have been removed.
- The `buzz.once()` function is used to schedule the buzzer functions (`buzzerRedRange` and `buzzerOrangeRange`) to run once without blocking the main loop.
- The `belowRedRange` and `belowOrangeRange` flags are updated based on the CO2 level and a hysteresis value (`BUZZER_HYSTERESIS`), ensuring smoother transitions.

This approach allows the main loop to continue executing while the buzzer functions play without introducing delays.

Please, test this (as I don't have a buzzer connected) and if it's working fine, we can change the code.

**Pull request merged**